src/main/java/seedu/address/model/ModelManager.java

```diff
@@ -12,35 +12,64 @@
 import seedu.address.commons.core.ComponentManager;
 import seedu.address.commons.core.LogsCenter;
 import seedu.address.commons.events.model.AddressBookChangedEvent;
+import seedu.address.commons.events.model.BookShelfChangedEvent;
+import seedu.address.model.book.Book;
+import seedu.address.model.book.exceptions.BookNotFoundException;
+import seedu.address.model.book.exceptions.DuplicateBookException;
 import seedu.address.model.person.Person;
 import seedu.address.model.person.exceptions.DuplicatePersonException;
 import seedu.address.model.person.exceptions.PersonNotFoundException;
 
 /**
- * Represents the in-memory model of the address book data.
+ * Represents the in-memory model of the book shelf data.
  * All changes to any model should be synchronized.
  */
 public class ModelManager extends ComponentManager implements Model {
     private static final Logger logger = LogsCenter.getLogger(ModelManager.class);
 
-    private final AddressBook addressBook;
-    private final FilteredList<Person> filteredPersons;
+    private final BookShelf bookShelf;
+    private final FilteredList<Book> filteredBooks;
+
+    private AddressBook addressBook;
+    private FilteredList<Person> filteredPersons;
+
+    /**
+     * Initializes a ModelManager with the given bookShelf and userPrefs.
+     */
+    public ModelManager(ReadOnlyBookShelf bookShelf, UserPrefs userPrefs) {
+        super();
+        requireAllNonNull(bookShelf, userPrefs);
+
+        logger.fine("Initializing with book shelf: " + bookShelf + " and user prefs " + userPrefs);
+
+        this.bookShelf = new BookShelf(bookShelf);
+        this.filteredBooks = new FilteredList<>(this.bookShelf.getBookList());
+        this.addressBook = new AddressBook();
```

Comment:
remove

Reply:
This is needed for now to keep tests from failing